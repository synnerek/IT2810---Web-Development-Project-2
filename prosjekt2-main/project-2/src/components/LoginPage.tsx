import axios from "axios";
import React from "react";

 
export default function LoginPage(){
    const checkApi=async ()=>{
      
      try{
          const testAxiosInstance = axios.create(
          {
          baseURL: "https://gitlab.stud.idi.ntnu.no/api/v4/projects/" + localStorage.getItem("id"),
          timeout: 3000,
          headers:{
              Authorization: "Bearer " + localStorage.getItem("token")
          }
          });
          const responseCode = await testAxiosInstance.get("")
          responseCode.status === 200 && 
          console.log("rsponskode", responseCode.status)
          sessionStorage.setItem("isLoggedIn", "true")  
          window.location.reload();
      }catch{
        sessionStorage.setItem("isLoggedIn", "false")
        console.log("Innlogging feilet, om dette vedvarer ring kundeservice, 95834337");
      }
    }


  return (
    <div className="app">
      <div className="login-container">
          <form className="form">
            <p>Please enter project ID and access key!</p>
            <input
              type="number"
              placeholder="Project ID"
              onChange={event => localStorage.setItem("id", event.target.value)}
            />
            <input
              type="password"
              placeholder="Access key"
              onChange={event => localStorage.setItem("token", event.target.value)}
            />
            <button className="authorButtons" type="button" onClick={checkApi}>Log in</button>
          </form>
      </div>
    </div>
  );
}





