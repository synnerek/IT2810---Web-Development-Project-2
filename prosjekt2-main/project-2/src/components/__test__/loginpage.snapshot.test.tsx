import React from 'react';
import renderer from 'react-test-renderer';
import LoginPage from '../LoginPage';

it('renders correctly', () => {
  const tree = renderer
    .create(<LoginPage />)
    .toJSON();
  expect(tree).toMatchSnapshot()
})

