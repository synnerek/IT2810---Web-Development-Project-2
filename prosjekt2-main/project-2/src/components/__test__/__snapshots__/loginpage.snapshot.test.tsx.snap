// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="App"
>
  <div
    className="login-container"
  >
    <form
      className="form"
    >
      <p>
        Please enter project ID and access key!
      </p>
      <input
        onChange={[Function]}
        placeholder="Project ID"
        type="number"
      />
      <input
        onChange={[Function]}
        placeholder="Access key"
        type="password"
      />
      <button
        className="authorButtons"
        onClick={[Function]}
        type="button"
      >
        Log in
      </button>
    </form>
  </div>
</div>
`;
